lcls_tools/common/devices/yaml/generate.py

```diff
@@ -409,3 +415,30 @@ def extract_lblms(self, area: Union[str, List[str]] = ["HTR"]):
             return complete_lblm_data
         else:
             return {}
+
+    def extract_metadata_by_device_names(
+        self,
+        device_names=Optional[List[str]],
+        required_fields=Optional[List[str]]
+    ):
+        # TODO: try not to call filter elements so many times as it parses csv
+        if required_fields:
+            elements = self._filter_elements_by_fields(
+                required_fields=required_fields
+            )
+        else:
+            elements = self._filter_elements_by_fields(
+                required_fields=self._required_fields
```

Comment:
Sorry, I'm not following the logic here? Maybe a Friday thing. So there is no case where something in `self._required_fields` would be in `required_fields`?